src/core/api/registry.go

```diff
@@ -38,6 +38,44 @@ func (t *RegistryAPI) Prepare() {
 	t.manager = ng.RegistryMgr
 }
 
+// Ping checks health status of a registry
+func (t *RegistryAPI) Ping() {
+	r := &model.Registry{}
+	t.DecodeJSONReqAndValidate(r)
+
+	var err error
+	id := r.ID
+	if id != 0 {
+		r, err = t.manager.Get(id)
+		if err != nil {
+			log.Errorf("failed to get registry %s: %v", r.Name, err)
+			t.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+			return
+		}
+
+		if r == nil {
+			t.CustomAbort(http.StatusNotFound, fmt.Sprintf("Registry %d not found", id))
+			return
+		}
+	}
+
+	if len(r.URL) == 0 || r.Credential == nil {
```

Comment:
I think the credential isn't necessary. For example, when I set up a replication policy that just pulling image from docker Hub.

Reply:
You are right, I will fix it.